lua/luasnip/init.lua

```diff
@@ -159,8 +159,15 @@ end
 local function jump(dir)
 	local current = session.current_nodes[vim.api.nvim_get_current_buf()]
 	if current then
-		session.current_nodes[vim.api.nvim_get_current_buf()] =
-			util.no_region_check_wrap(safe_jump_current, dir)
+		local next_node = util.no_region_check_wrap(safe_jump_current, dir)
+
+		if session.config.exit_roots then
+			if next_node.pos == 0 and next_node.parent.parent_node == nil then
```

Comment:
Ah, I checked the failing test: it's possible that next_node itself is nil (happens when a jump caused some error, then `safe_jump_current` may return nil), could you handle that here? Just making sure it's non-nil before accessing .pos should be enough :)

Reply:
I checked the problem, and tried to implement what you suggested, but this did not help with tests. Removing exit_roots implementation allows tests to pass. I suspect if test needs to be updated. Tomorrow, I will check out the tests manually, then I will report back.

Reply:
Some tests also rely on exit_roots=false, passing that in addition should make them pass

Reply:
Is returning `false` and setting `current_node` to `nil` if we have no `next_node` OK?

Reply:
I think we currently return `true` because there was a jump.. I think we'd best stick to that